rosbridge_library/src/rosbridge_library/internal/subscribers.py

```diff
@@ -100,16 +101,28 @@ def __init__(self, topic, client_id, callback, node_handle, msg_type=None, raw=F
         if topic_type is not None and topic_type != msg_type_string:
             raise TypeConflictException(topic, topic_type, msg_type_string)
 
+        # Get publishers info
+        publishers_info = node_handle.get_publishers_info_by_topic(topic)
+
+        # Select QoS
+        default_qos_profile = 10
+        if publishers_info:
+            qos_profile = publishers_info[0].qos_profile
+            # Set history for Foxy
+            if qos_profile.history == HistoryPolicy.UNKNOWN:
+                qos_profile.history = HistoryPolicy.KEEP_LAST
+        else:
+            qos_profile = default_qos_profile
```

Comment:
```suggestion
        if publishers_info:
            qos_profile = publishers_info[0].qos_profile
            qos_profile.history = HistoryPolicy.KEEP_LAST
            qos_profile.depth = 10
        else:
            qos_profile = default_qos_profile
```

Reply:
@emersonknapp  I'm sorry to be late, I had to overwrite all the members that have `Duration` Class to resolve errors in `Foxy` and `rmw_fastrtps`.
It seemed to be set to `INT_MAX` or something, but if created by int it was `0`.

Do you know anything about this and do you have some advice for the implementation approach?
1038fba845f5deb1b314168198bd2d0c405220cf